src/KioskMode/README.md

```diff
@@ -0,0 +1,102 @@
+# Kiosk-Mode
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”,
+“SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be
+interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+**Table of Contents**
+* [Purpose](#purpose)
+* [Architecture](#architecture)
+* [Implementing a Provider](#implementing-a-provider)
+* [Implementing a Player](#implementing-a-player)
+
+## Purpose
+
+A kiosk mode is understood to be a special mode to display certain applications, or
+in the context of ILIAS, certain views in an application where the user gets less
+control over the application or feature than usual. This is traditionally used for
+the unsupervised operation of the application at public locations like museums or
+train stations, where the user e.g. should not be able to perform otherwise common
+operations like closing a window or shutting down the system.
+
+Similar requirements arise in ILIAS for different objects and different scenarios:
+
+* A Test needs to be displayed in a restricted way for the use in an E-Exam.
+* The LTI-Interface needs the ability to display objects in a Kiosk-Mode when it
+  ILIAS is used as a Tool-Provider.
+* The Learning Sequence requires a kiosk mode to maintain control over the general
+  display and navigation.
```

Comment:
After the discussion yesterday I think the "maintain control" requirements differ greatly between these scenarios. The current LTI support in learning modules e.g. adds just a view lines of code in the existing presentation view. The existing views can basically be re-used in their current state. I agree that the current (LTI) implementation needs a better base implementation but I would favour a separation of the basic kiosk mode requirements (focus on single object presentation, i.e. basically do not show the main menu) and the learning sequence requirements (take over full control of navigation).

Reply:
I think you are confusing implementation and requirement here. As far as I remember the discussion, the LTI-implementation was driven by the mere impossibility to implement another solution besides masking selected parts of GUIs to implement the requirement that most of ILIAS should just work (tm) in LTI. Root cause to these problems IMO is, that there is little required or encoded commonality in the way GUI-classes are implemented in ILIAS, which makes it hard to reuse them on a layer below rendering. This attempts to partly fix this, so at least a content centric kiosk mode for the learning sequence (and possibly other features) won't need to pull the same brittle magic tricks again.